### PR TITLE
fix(logging): Support numbers when unmarshaling Severity

### LINKED ISSUE #4215: logging: log.Entry.Severity can't unmarshal after marshalling
**Client**
logging

**Environment**
MacOS Mojave

**Go Environment**
```
go version go1.15.5 darwin/amd64
GOARCH="amd64"
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.15.5/libexec"
GOSUMDB="sum.golang.org"
GOTOOLDIR="/usr/local/Cellar/go/1.15.5/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/__/kst87n5n6xs1wr9vn3gw9xxh0000gn/T/go-build111216510=/tmp/go-build -gno-record-gcc-switches -fno-common"
```
**Code**
```go
package main

import (
	"fmt"
	"encoding/json"
	"cloud.google.com/go/logging"
)

func main() {
	entry := logging.Entry { Payload: "test log", Severity: logging.Info}
	j, err := json.Marshal(entry)
	if err != nil {
		panic(err)
	}

	var entryU logging.Entry
	err = json.Unmarshal(j, &entryU)
	if err != nil {
		panic(err)
	}
}
```

**Expected behavior**
Marshaling then Unmarshalling a logging.Entry struct should produce the same struct without errors.

**Actual behavior**
Unmarshalling has an error
```
panic: json: cannot unmarshal number into Go struct field Entry.Severity of type string
```

This is due to `logging.Entry.Severity` having an `UnmarshalJSON` method that only parses `string` `Severity` but during Marshaling, `Severity` is written as a number since it's underlying type is `int`

**Additional context**
This occurred while I was trying to process logs from `logging/logadmin` that had been saved as JSON, assuming Marshalling/Unmarshalling would be symmetrical.

### PR DESCRIPTION
**Pull request by alutman**
logging.Entry.Severity is written out as a JSON number during
Marshalling but has an UnmarshalJSON method that only supports Severity
as as string.

This commit adds support to unmarshal JSON numbers in addition to
strings so Marshalling then Unmarshalling the same object works without
error

Fixes #4215

### CODE REVIEW

**Review by nicoleczhu — APPROVED** [review]

**Comment by nicoleczhu** [comment]
Thanks for this!

**Comment by nicoleczhu** [comment]
@alutman do you want us to merge this - is it ready?

With how CI is configured and given this repo has a lot of PRs updating changes, you mgiht be stuck in a cycle of having to update this branch, and then wait for someone internal to approve HEAD, in order for CI to run, before it can be merged 

Let us know if this PR is ready, and I can merge for you.

**Comment by alutman** [comment]
@nicoleczhu 
Yes please, merge it. It's ready.

**Pull request merged**
